libnetwork/drivers/bridge/interface_linux_test.go

```diff
@@ -16,35 +44,142 @@ func TestInterfaceDefaultName(t *testing.T) {
 	}
 	config := &networkConfiguration{}
 	_, err = newInterface(nh, config)
-	if err != nil {
-		t.Fatalf("newInterface() failed: %v", err)
-	}
+	assert.NilError(t, err)
+	assert.Equal(t, config.BridgeName, DefaultBridgeName)
+}
 
-	if config.BridgeName != DefaultBridgeName {
-		t.Fatalf("Expected default interface name %q, got %q", DefaultBridgeName, config.BridgeName)
-	}
+func TestAddressesNoInterface(t *testing.T) {
+	i := bridgeInterface{}
+	addrs, err := i.addresses(netlink.FAMILY_V6)
+	assert.NilError(t, err)
+	assert.Check(t, addrs != nil)
+	assert.Check(t, len(addrs) == 0)
```

Comment:
Mostly curious here; in many situations, an empty slice is "roughly" identical to a `nil` slice;  e.g., the following passes;

```go
func foo() []string {
	return nil
}

func TestFoo(t *testing.T) {
	bla := foo()
	assert.Assert(t, is.Len(bla, 0))
}
```

For our use, is there an important distinction between the two? (empty slice vs `nil`)?

Reply:
> [When designing interfaces, avoid making a distinction between a nil slice and a non-nil, zero-length slice, as this can lead to subtle programming errors.](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices:~:text=When%20designing%20interfaces%2C%20avoid%20making%20a%20distinction%20between%20a%20nil%20slice%20and%20a%20non%2Dnil%2C%20zero%2Dlength%20slice%2C%20as%20this%20can%20lead%20to%20subtle%20programming%20errors.)